### PR TITLE
Fix predict with confidence interval

### PR DESCRIPTION
**Pull request by nalimilan**
DataFrames aren't a dependency anymore.

Fixes https://github.com/JuliaStats/StatsModels.jl/issues/159. Covered by tests at https://github.com/JuliaStats/GLM.jl/pull/335 (we should add tests with a dummy model type at some point).

### CODE REVIEW

File: src/statsmodel.jl [hunk 1]
```diff
@@ -131,26 +131,26 @@ StatsBase.adjr2(mm::TableRegressionModel) = adjr2(mm.model)
 StatsBase.r2(mm::TableRegressionModel, variant::Symbol) = r2(mm.model, variant)
 StatsBase.adjr2(mm::TableRegressionModel, variant::Symbol) = adjr2(mm.model, variant)
 
-function _return_predictions(yp::AbstractVector, nonmissings, len)
+function _return_predictions(T, yp::AbstractVector, nonmissings, len)
     out = missings(eltype(yp), len)
     out[nonmissings] = yp
     out
```
Comment (nalimilan) [inline src/statsmodel.jl]: Note that in this case, we return a mere vector (not a data frame) even if the input is a data frame. I guess that's the most logical output.

Comment (kleinschmidt) [reply src/statsmodel.jl]: I think that's what the consensus was right?  I remember their being some extensive discussion about this...

**Comment by kleinschmidt** [comment]
Weird how this snuck in; I guess the predict PRs were opened while #71 was still being finalized and didn't touch any of the same code so I didn't notice.  We really ought to have tests for this though, could you add some?

File: src/statsmodel.jl (hunk 1 above)
Comment (nalimilan) [reply src/statsmodel.jl]: Ah yes that was https://github.com/JuliaStats/StatsModels.jl/pull/81.

**Comment by nalimilan** [comment]
> Weird how this snuck in; I guess the predict PRs were opened while #71 was still being finalized and didn't touch any of the same code so I didn't notice.

Yes that seems to be what happened with https://github.com/JuliaStats/StatsModels.jl/pull/81.

> We really ought to have tests for this though, could you add some?

 JuliaStats/GLM.jl#335 adds tests in GLM. I'll have a look at adding tests here.

**Comment by nalimilan** [comment]
OK, tests added!

**Comment by kleinschmidt** [comment]
Oh and bump the patch version?

**Comment by nalimilan** [comment]
Done.

**Pull request merged**